Consolidate duplicate `get_retrievable_type` overrides into the UPE base class (#5053)
woocommerce/woocommerce-gateway-stripe: changelog.txt

* Remove - Remove giropay from new checkouts (deprecated by Stripe on 2024-06-30); legacy refund and past-order rendering preserved
* Dev - Exclude AGENTS.md and CLAUDE.md contributor-instruction files from the built plugin zip
* Add - Append a "what's new" changelog link to the Updated! message after manually updating the plugin from the Plugins page

> * Dev - Consolidate duplicate `get_retrievable_type` and `is_available_for_account_country` implementations into `WC_Stripe_UPE_Payment_Method`; non-default implementations (Bancontact, iDEAL, OC, Sofort, Amazon Pay, Link) are retained

Reviewer: I think we should simplify the dev entry.
```suggestion
* Dev - Remove redundant `get_retrievable_type()` and `is_available_for_account_country()` overrides from payment methods
```

Reviewer 2: Applied verbatim.

> * Fix - Hide Boleto, OXXO, and ACSS at checkout when the merchant's Stripe account country does not support them

Reviewer: Based on our code, this actually fixes the admin experience as well.
```suggestion
* Fix - Hide Boleto, OXXO, and ACSS at checkout and in the admin UI when they are not available to the merchant
```

Reviewer 2: Applied verbatim.
